modules/core/handler_modules.php

```diff
@@ -961,3 +961,149 @@ function warn_for_unsaved_changes_callback($val) {
+           'srv_setup_stepper_email',
+           'srv_setup_stepper_password',
+           'srv_setup_stepper_provider',
+           'srv_setup_stepper_is_sender',
+           'srv_setup_stepper_is_receiver',
+           'srv_setup_stepper_smtp_address',
+           'srv_setup_stepper_smtp_port',
+           'srv_setup_stepper_smtp_tls',
+           'srv_setup_stepper_imap_address',
+           'srv_setup_stepper_imap_port',
+           'srv_setup_stepper_imap_tls',
+           'srv_setup_stepper_enable_sieve',
+           'srv_setup_stepper_create_profile',
+           'srv_setup_stepper_profile_is_default',
+           'srv_setup_stepper_profile_signature',
+           'srv_setup_stepper_profile_reply_to',
+           'srv_setup_stepper_imap_sieve_host',
+           'srv_setup_stepper_only_jmap',
+           'srv_setup_stepper_jmap_hide_from_c_page',
+           'srv_setup_stepper_jmap_address',
+           ));
+
+        if ($success) {
+            // Destructure form array into variables
+            [
+                'srv_setup_stepper_profile_name' => $profileName,
+                'srv_setup_stepper_email' => $email,
+                'srv_setup_stepper_password' => $password,
+                'srv_setup_stepper_provider' => $provider,
+                'srv_setup_stepper_is_sender' => $isSender,
+                'srv_setup_stepper_is_receiver' => $isReceiver,
+                'srv_setup_stepper_smtp_address' => $smtpAddress,
+                'srv_setup_stepper_smtp_port' => $smtpPort,
+                'srv_setup_stepper_smtp_tls' => $smtpTls,
+                'srv_setup_stepper_imap_address' => $imapAddress,
+                'srv_setup_stepper_imap_port' => $imapPort,
+                'srv_setup_stepper_imap_tls' => $imapTls,
+                'srv_setup_stepper_enable_sieve' => $enableSieve,
+                'srv_setup_stepper_create_profile' => $createProfile,
+                'srv_setup_stepper_profile_is_default' => $profileIsDefault,
+                'srv_setup_stepper_profile_signature' => $profileSignature,
+                'srv_setup_stepper_profile_reply_to' => $profileReplyTo,
+                'srv_setup_stepper_imap_sieve_host' => $imapSieveHost,
+                'srv_setup_stepper_only_jmap' => $onlyJmap,
+                'srv_setup_stepper_jmap_hide_from_c_page' => $jmapHideFromCPage,
+                'srv_setup_stepper_jmap_address' => $jmapAddress,
+            ] = $form;
+            
+            /*
+            * When JMAP selected only configure JMAP
+            */
+             if(isset($onlyJmap) && $onlyJmap) {
+                if (!$this->module_is_supported('jmap')) {
+                    Hm_Msgs::add("ERRJMAP module is not enabled");
+                    return;
+                }
+                
+                $this->jmap_server_id = connect_to_jmap_server($jmapAddress, $jmapHideFromCPage, $profileName, $email, $password, 'imap', $this);
+                 connect_to_imap_server(
```

Comment:
Isn't this a leftover that should be removed? Why do we call connect to jmap and imap server simultaneously?